nimbus-core/src/main/java/com/antheminc/oss/nimbus/converter/excel/UnivocityExcelToCSVConverter.java

```diff
@@ -0,0 +1,127 @@
+/**
+ *  Copyright 2016-2019 the original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.antheminc.oss.nimbus.converter.excel;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.IntStream;
+
+import org.apache.commons.lang.RandomStringUtils;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.poi.EncryptedDocumentException;
+import org.apache.poi.openxml4j.exceptions.InvalidFormatException;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.CellType;
+import org.apache.poi.ss.usermodel.DataFormatter;
+import org.apache.poi.ss.usermodel.FormulaEvaluator;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.ss.usermodel.WorkbookFactory;
+
+import com.antheminc.oss.nimbus.FrameworkRuntimeException;
+import com.univocity.parsers.csv.CsvWriter;
+import com.univocity.parsers.csv.CsvWriterSettings;
+
+/**
+ * <p>Uses Univocity to convert excel files to their .csv equivalent and stores
+ * that file in the local filesystem's temporary storage.
+ * 
+ * @author Tony Lopez
+ * @author Sandeep Mantha
+ */
+public class UnivocityExcelToCSVConverter implements ExcelToCsvConverter {
+
+	@Override
+	public List<File> convert(InputStream inputStream, ExcelParserSettings settings) throws IOException {
+		try (Workbook workbook = WorkbookFactory.create(inputStream)) {
+			return convert(workbook, settings);
+		} catch (EncryptedDocumentException | InvalidFormatException e) {
+			throw new FrameworkRuntimeException("An error occurred while conerting the Excel workbook object.", e);
```

Comment:
Can the name of the Workbook be included within the exception message?

Reply:
No, but if an `IOException` is thrown, the exception would bubble up and be caught by the `WebActionController`, where a more meaningful exception message will be provided.